autopilot/ipfilter.go

```diff
@@ -20,29 +21,86 @@ const (
 	resolverLookupTimeout = 5 * time.Second
 )
 
+var (
+	errNoSuchHost        = errors.New("no such host")
+	errTooManyAddresses  = errors.New("host has more than two addresses, or two of the same type")
+	errUnparsableAddress = errors.New("host address could not be parsed to a subnet")
+)
+
 type resolver interface {
 	LookupIPAddr(ctx context.Context, host string) ([]net.IPAddr, error)
 }
 
 type ipFilter struct {
-	subnets  map[string]string
-	resolver resolver
-	timeout  time.Duration
+	hostIPToSubnetsCache map[string][]string
```

Comment:
I think this is reasonable but I'm honestly surprised that it's necessary. The operating system should cache dns lookups instead of returning a `server misbehaving` every now and then.